### PR TITLE
DOC: Tidy Tools Docstrings

### PR DESCRIPTION
**Pull request by John-P**
- Wrap Lines
- Fix typos
- Make indentation consistent

### CODE REVIEW

File: tiatoolbox/tools/graph.py [hunk 1]
```diff
@@ -44,12 +44,15 @@ def delaunay_adjacency(points: ArrayLike, dthresh: Number) -> list:
     See https://en.wikipedia.org/wiki/Adjacency_matrix.
 
     Args:
-        coordinates (ArrayLike): A nxm list of coordinates.
-        dthresh (int): Distance threshold for triangulation.
+        coordinates (ArrayLike):
```
Comment (shaneahmed) [inline tiatoolbox/tools/graph.py:47]:
```suggestion
        points(ArrayLike):
```

File: tiatoolbox/tools/patchextraction.py [hunk 2]
```diff
@@ -284,30 +307,37 @@ def filter_coordinates_fast(
...
+                A virtual pyramidal reader of the mask related to the
+                WSI from which we want to extract the patches.
+            coordinates_list (ndarray and np.int32):
+                Coordinates to be checked via the `func`. They must be
+                in the same resolution as requested `resolution` and
+                `units`. The shape of `coordinates_list` is (N, K) where
+                N is the number of coordinate sets and K is either 2 for
+                centroids or 4 for bounding boxes. When using the
+                default `func=None`, K should be 4, as we expect the
+                `coordinates_list` to be refer to bounding boxes in
```
Comment (shaneahmed) [inline tiatoolbox/tools/patchextraction.py:325]:
```suggestion
                `coordinates_list` of bounding boxes in
```

File: tiatoolbox/tools/patchextraction.py [hunk 3]
```diff
@@ -349,39 +379,51 @@ def get_coordinates(
...
-            image_shape (a tuple (int, int) or :class:`numpy.ndarray` of shape (2,)):
-                This argument specifies the shape of mother image (the image we want to
-                extract patches from) at requested `resolution` and `units` and it is
-                expected to be in (width, height) format.
-            patch_input_shape (a tuple (int, int) or
-                :class:`numpy.ndarray` of shape (2,)): Specifies the input shape of
-                requested patches to be extracted from mother image at desired
-                `resolution` and `units`. This argument is also expected to be in
+            image_shape (a tuple (int, int) or :class:`numpy.ndarray` of
+            shape (2,)):
```
Comment (shaneahmed) [inline tiatoolbox/tools/patchextraction.py:383]: No input parameter `shape`

File: tiatoolbox/tools/patchextraction.py [hunk 4]
```diff
@@ -349,39 +379,51 @@ def get_coordinates(
...
+                the patches with output location exceeds the
+                `image_shape` would be neglected. Otherwise, those
+                patches would be extracted with `Reader` function and
+                appropriate padding.
 
         Return:
-            coord_list: a list of coordinates in `[start_x, start_y, end_x, end_y]`
-            format to be used for patch extraction.
+            coord_list:
+                A list of corrdinates in `[start_x, start_y, end_x,
```
Comment (shaneahmed) [inline tiatoolbox/tools/patchextraction.py:425]:
```suggestion
                A list of coordinates in `[start_x, start_y, end_x,
```

File: tiatoolbox/tools/pyramid.py [hunk 5]
```diff
@@ -18,15 +18,15 @@
...
 method.
 
 These are generally intended for serialisation or streaming via a web
-UI. The `get_tile` method returns a Pillow Image object which can
-be easily serialised via the use of an io.BytesIO object or saved
-directly to disk.
+UI. The `get_tile` method returns a Pillow Image object which can be
+easily serialised via the use of an io.BytesIO object or saved directly
+to disk.
 """
```
Comment (shaneahmed) [inline tiatoolbox/tools/pyramid.py:30]: new line?

File: tiatoolbox/tools/pyramid.py [hunk 6]
```diff
@@ -52,15 +52,18 @@ class TilePyramidGenerator:
...
-            Default is 2.
-        tile_overlap (int): The number of extra pixel to add to each
-            edge of the tile. Default is 0.
+        tile_size (int):
+            The size of tiles to generate. Default is 256. Note that the
+            output tile size will be :math:`\text{tile size} + 2
+            \times\text{overlap}`.
+        downsample (int):
+            The downsample factor between levels. Default is 2.
+        tile_overlap (int):
```
Comment (shaneahmed) [inline tiatoolbox/tools/pyramid.py:64]: not defined in `__init__`

File: tiatoolbox/tools/pyramid.py [hunk 7]
```diff
@@ -170,28 +175,27 @@ def get_tile(
...
+                slide in one tile, 0-0-0).
             x (int):
                 The tile index in the x direction.
             y (int):
                 The tile index in the y direction.
             pad_mode (str):
-                Method for padding when reading areas outside of
-                the input image. Default is constant (0 padding). This
-                is passed to `read_func` which defaults to
+                Method for padding when reading areas outside of the
```
Comment (shaneahmed) [inline tiatoolbox/tools/pyramid.py:185]:
```suggestion
                Method for padding when reading area is outside of the
```

File: tiatoolbox/tools/pyramid.py [hunk 8]
```diff
@@ -386,15 +394,14 @@ class ZoomifyGenerator(TilePyramidGenerator):
             The WSI reader object. Must implement
             `tiatoolbox.wsicore.wsi_Reader.WSIReader.read_rect`.
         tile_size (int):
```
Comment (shaneahmed) [inline tiatoolbox/tools/pyramid.py:396]: are these variables initialised? may be using super()?

File: tiatoolbox/tools/scale.py [hunk 9]
```diff
@@ -59,12 +61,15 @@ def __init__(self, num_iters=100):
...
         Args:
-            logits (array-like): Classifier output scores.
-            labels (array like): Classifier labels, must be `+1` vs `-1` or `1` vs `0`.
+            logits (array-like):
+                Classifier output scores.
+            labels (array like):
+                Classifier labels, must be `+1` vs `-1` or `1` vs `0`.
+
         Returns:
             Model with fitted coefficients a and b for the sigmoid function.
```
Comment (shaneahmed) [inline tiatoolbox/tools/scale.py:75]: line 79 epsilon

File: tiatoolbox/tools/scale.py [hunk 10]
```diff
@@ -168,9 +173,12 @@ def transform(self, logits):
         """Tranform input to probabilities basing on trained parameters.
```
Comment (shaneahmed) [inline tiatoolbox/tools/scale.py:173]:
```suggestion
        """Transform input to probabilities basing on trained parameters.
```

File: tiatoolbox/tools/scale.py [hunk 11]
```diff
@@ -168,9 +173,12 @@ def transform(self, logits):
...
 
         Args:
-            labels (array like): Classifier labels, must be `+1` vs `-1` or `1` vs `0`.
+            labels (array like):
```
Comment (shaneahmed) [inline tiatoolbox/tools/scale.py:176]:
```suggestion
            logits (array like):
```

File: tiatoolbox/tools/scale.py [hunk 12]
```diff
@@ -179,10 +187,14 @@ def fit_transform(self, logits, labels):
         """Fit and tranform input to probabilities.
```
Comment (shaneahmed) [inline tiatoolbox/tools/scale.py:187]:
```suggestion
        """Fit and transform input to probabilities.
```

File: tiatoolbox/tools/stainaugment.py [hunk 13]
```diff
@@ -35,49 +35,66 @@
...
+        n_stains (int):
+            Number of stain channels in the stain concentrations.
             Expected to be 2 for H&E stained images.
-        tissue_mask (:class:`numpy.ndarray`): tissue region mask in the image.
+        tissue_mask (:class:`numpy.ndarray`):
+            Tissue region mask in the image.
 
     Examples:
         >>> '''Using the stain augmentor in the 'albumentations' pipeline'''
         >>> from tiatoolbox.tools.stainaugment import StainAugmentaiton
```
Comment (shaneahmed) [inline tiatoolbox/tools/stainaugment.py:95]:
```suggestion
        >>> from tiatoolbox.tools.stainaugment import StainAugmentor
```

File: tiatoolbox/tools/stainaugment.py [hunk 14]
```diff
@@ -91,7 +108,7 @@ class StainAugmentor(ImageOnlyTransform):
 
         >>> '''Using the stain augmentor stand alone'''
         >>> from tiatoolbox.tools.stainaugment import StainAugmentaiton
```
Comment (shaneahmed) [inline tiatoolbox/tools/stainaugment.py:110]:
```suggestion
        >>> from tiatoolbox.tools.stainaugment import StainAugmentor
```

File: tiatoolbox/tools/stainaugment.py [hunk 15]
```diff
@@ -166,13 +187,15 @@ def fit(self, img, threshold=0.85):
     def augment(self):
         """Return an augmented instance based on source stain concentrations.
 
-        Stain concentrations of the source image are altered (scaled and shifted)
-        based on the random alpha and beta paramters, and then an augmented image is
-        reconstructed from the altered concentrations.
-        All parameters needed for this part are calculated when calling fit() function.
+        Stain concentrations of the source image are altered (scaled and
+        shifted) based on the random alpha and beta paramters, and then
```
Comment (shaneahmed) [inline tiatoolbox/tools/stainaugment.py:191]:
```suggestion
        shifted) based on the random alpha and beta parameters, and then
```

File: tiatoolbox/tools/tissuemask.py [hunk 16]
```diff
@@ -110,10 +113,11 @@ def fit(self, images: np.ndarray, masks=None) -> None:
...
         Args:
-            images (:class:`numpy.ndarray`): List of images with a
-              length 4 shape (N, height, width, channels).
-            masks (:class:`numpy.ndarray`): Unused here, for API
-              consistency.
+            images (:class:`numpy.ndarray`):
+                List of images with a length 4 shape (N, height, width,
+                channels).
+            masks (:class:`numpy.ndarray`):
+                Unused here, for API consistency.
```
Comment (shaneahmed) [inline tiatoolbox/tools/tissuemask.py:120]: line 127

```suggestion
                f" but received shape of 
```

**Review by shaneahmed — CHANGES_REQUESTED** [review]
Please address the changes as mentioned.

File: tiatoolbox/tools/pyramid.py (hunk 8 above)
Comment (John-P) [reply tiatoolbox/tools/pyramid.py:397]: Yes, they are initialised in the parent class init (the init is inherited unmodified).

**Comment by John-P** [comment]
@shaneahmed  Thank you very much for your helpful feedback. I have addressed all of your comments now.

**Review by shaneahmed — APPROVED** [review]

**Review by vqdang — APPROVED** [review]

**Pull request merged**